src/main/java/edu/kit/ipd/crowdcontrol/objectservice/ExperimentController.java

```diff
@@ -0,0 +1,120 @@
+import edu.kit.ipd.crowdcontrol.objectservice.crowdworking.TaskOperationException;
+import edu.kit.ipd.crowdcontrol.objectservice.event.ChangeEvent;
+import edu.kit.ipd.crowdcontrol.objectservice.event.Event;
+import edu.kit.ipd.crowdcontrol.objectservice.event.EventManager;
+import edu.kit.ipd.crowdcontrol.objectservice.event.EventObservable;
+import edu.kit.ipd.crowdcontrol.objectservice.proto.Experiment;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import rx.Observable;
+import rx.Observer;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by lucaskrauss at 05.02.2016
+ */
+public class ExperimentController implements Observer<Event<Experiment>> {
+
+    private final Logger log = LogManager.getLogger(ExperimentController.class);
+    private final Observable<Event<Experiment>> observable = EventManager.EXPERIMENT_CREATE.getObservable();
+    private final EventObservable<ChangeEvent<Experiment>> endExpObservable = EventManager.EXPERIMENT_CHANGE;
+    private final PlatformManager platformManager;
+
+    public ExperimentController(PlatformManager manager){
+        platformManager = manager;
+        observable.subscribe();
+
+    }
+
+
+    /**
+     * Starts the experiment by publishing it on the participating platforms.
+     * If an error occurs during the publishing, all already created tasks for this experiment will be undone
+     * @param experiment to be started
+     */
+   private void startExperiment(Experiment experiment){
+
+        for (int i = 0; i < experiment.getPopulationsCount(); i++) {
+            try {
+                platformManager.publishTask(experiment.getPopulations(i).getPlatformId(),experiment);
+
+            }catch ( TaskOperationException e1){
+                //could not create task
+                log.fatal(String.format("Error! Could not create experiment on platform %s!", experiment.getPopulations(i).getPlatformId()), e1);
+                unpublishExperiment(experiment);
+            }catch (IllegalStateException | IllegalArgumentException e2){
+                log.fatal("Error! Could not create experiment! "+e2.getMessage());
+                unpublishExperiment(experiment);
+            }
+        }
+
+        //ChangeEvent mit State.INVALID ?
+
+
+    }
+
+    /**
+     * Unpublishes an experiment from all platforms it is meant to be active on.
+     * This method is only called if the initialization of the experiment has failed on one
+     * of its platforms.
+     * @param experiment which is going to be unpublished
+     */
+    private void unpublishExperiment(Experiment experiment){
+        for (int i = 0; i < experiment.getPopulationsCount(); i++) {
+            try {
+                platformManager.unpublishTask(experiment.getPopulations(i).getPlatformId(),experiment);
+            }catch ( TaskOperationException e1){
+                //nothing to do here because experiment-publishing was aborted before this population was reached.
```

Comment:
just log the exception. This can also be evil. since platforms will still send answers...